lib/order.rb

```diff
@@ -0,0 +1,44 @@
+require_relative 'menu'
+require_relative 'notifier'
+
+class Order
+
+  attr_reader :basket, :menu
+
+  def initialize(menu = Menu.new)
+    @basket = Hash.new(0)
+    @menu = menu
+  end
+
+  def add_item(item, quantity = 1)
+    check_item_exits(item)
+    @basket[item] = @basket[item] + quantity
+    add_basket_confirmation(item, quantity)
+  end
+
+  def total_order    
```

Comment:
Overall very impressive code. Would be good to see a README showing how it all works together